Find a functionally equivalent code:android.media.AudioService.makeA2dpDeviceUnavailableNow:COMMENT
Method Modifier: private     
Comment:// must be called synchronized on mConnectedDevices

@@ -4,7 +4,14 @@
     }
     AudioSystem.setDeviceConnectionState(AudioSystem.DEVICE_OUT_BLUETOOTH_A2DP, AudioSystem.DEVICE_STATE_UNAVAILABLE, address);
     mConnectedDevices.remove(AudioSystem.DEVICE_OUT_BLUETOOTH_A2DP);
+    synchronized (mCurAudioRoutes) {
+        // Remove A2DP routes as well
+        if (mCurAudioRoutes.mBluetoothName != null) {
+            mCurAudioRoutes.mBluetoothName = null;
+            sendMsg(mAudioHandler, MSG_REPORT_NEW_ROUTES, SENDMSG_NOOP, 0, 0, null, 0);
+        }
+    }
 }

